src/Umbraco.Web.UI.Client/src/common/services/clipboard.service.js

```diff
@@ -1,203 +1,276 @@
-        
-        saveStorage(storage);
-    };
-    
-    
-    
-    return service;
-}


-angular.module("umbraco.services").factory("clipboardService", clipboardService);

+/**
+ * @ngdoc service
+ * @name umbraco.services.clipboardService
+ *
+ * @requires notificationsService
+ * @requires eventsService
+ *
+ * @description
+ * Service to handle clipboard in general across the application. Responsible for handling the data both storing and retrive.
+ * The service has a set way for defining a data-set by a entryType and alias, which later will be used to retrive the posible entries for a paste scenario.
+ *
+ */
+function clipboardService(notificationsService, eventsService, localStorageService, iconHelper) {
+    
+    
+    var STORAGE_KEY = "umbClipboardService";
+    
+    var retriveStorage = function() {
+        if (localStorageService.isSupported === false) {
+            return null;
+        }
+        var dataJSON;
+        var dataString = localStorageService.get(STORAGE_KEY);
+        if (dataString != null) {
+            dataJSON = JSON.parse(dataString);
+        }
+        
+        if(dataJSON == null) {
+            dataJSON = new Object();
+        }
+        
+        if(dataJSON.entries === undefined) {
+            dataJSON.entries = [];
+        }
+        
+        return dataJSON;
+    }
+    
+    var saveStorage = function(storage) {
+        var storageString = JSON.stringify(storage);
+        
+        try {
+            var storageJSON = JSON.parse(storageString);
+            localStorageService.set(STORAGE_KEY, storageString);
+            
+            eventsService.emit("clipboardService.storageUpdate");
+            
+            return true;
+        } catch(e) {
+            return false;
+        }
+        
+        return false;
+    }
+
+    var prepareEntryForStorage = function(entryData) {
+
+        var shallowCloneData = Object.assign({}, entryData);// Notice only a shallow copy, since we dont need to deep copy. (that will happen when storing the data)
+        delete shallowCloneData.key;
+        delete shallowCloneData.$$hashKey;
+        
+        return shallowCloneData;
+    }
+    
+    
+    var service = {};
+    
+    /**
+    * @ngdoc method
+    * @name umbraco.services.clipboardService#copy
+    * @methodOf umbraco.services.clipboardService
+    *
+    * @param {string} type A string defining the type of data to storing, example: 'elementType', 'contentNode'
+    * @param {string} alias A string defining the alias of the data to store, example: 'product'
+    * @param {object} entry A object containing the properties to be saved, this could be the object of a ElementType, ContentNode, ...
+    * @param {string} displayLabel (optional) A string swetting the label to display when showing paste entries.
+    *
+    * @description
+    * Saves a single JS-object with a type and alias to the clipboard.
+    */
+    service.copy = function(type, alias, data, displayLabel) {
```

Comment:
Method/function arguments changed do we know if anyone used this in their own packages as this would be considered a breaking change @nielslyngsoe

Reply:
No worries